### ISSUE TITLE
[Feature] Add Voting system

### ISSUE THREAD

**Issue by ManishBisht777** [rupali-codes/LinksHub#495]
### Description

It would be good if we can add some voting functionality or some sort of ratings that can help in identifying the best one to choose between a list of Resources

but I think rating might be a incorrect way to choose resources as thats comes down to preference

Lets see what other think of it

**Comment by CBID2** [comment]
> ### Description
> It would be good if we can add some voting functionality or some sort of ratings that can help in identifying the best one to choose between a list of Resources
> 
> but I think rating might be a incorrect way to choose resources as thats comes down to preference
> 
> Lets see what other think of it
> 
> ### Screenshots
> _No response_
> 
> ### Additional Information
> _No response_

Interesting idea. While it would create more engagement, I feel like a con for the voting system would be that people use it to center certain sources due to the creator’s and topic’s popularity not so much of how they convey information.

**Comment by rupali-codes** [comment]
Agree with @CBID2 and there's also some really good but Underrated content which people are not aware of,

**Comment by Weverak** [comment]
When you click on name of site it opens some kind of windows with name, URL and some description. Maybe it's good place for it. It might be done as let's say 3 rating categories and these three values could be rounded and then we can save the rounded value for each answer and round it too. So like this:

Category 1: 7/10
Category 2: 3/10
Category 3: 9/10

19/3 equals 6.333333 this value would be saved and all responses would be rounded.

**Comment by rupali-codes** [comment]
> When you click on name of site it opens some kind of windows with name, URL and some description. Maybe it's good place for it. It might be done as let's say 3 rating categories and these three values could be rounded and then we can save the rounded value for each answer and round it too. So like this:
> 
> Category 1: 7/10 Category 2: 3/10 Category 3: 9/10
> 
> 19/3 equals 6.333333 this value would be saved and all responses would be rounded.

Sounds good, but as of now we are not using any kind of authentication, so people/users gonna vote for any site particularly ?

**Comment by pramodreddypandiri** [comment]
@rupali-codes , How about maintain click count and put it somewhere on card.

**Comment by Weverak** [comment]
We could use cookies. Users don't delete them usually and you can do it easily, but cookies will expire someday. Maybe firebase authentication, but users would have to register and that's another account for them and it's not big deal at all. So maybe this could be the way.

@pramodreddypandiri clicks aren't bad, but where could we maintain them.?

**Comment by rupali-codes** [comment]
> We could use cookies. Users don't delete them usually and you can do it easily, but cookies will expire someday. Maybe firebase authentication, but users would have to register and that's another account for them and it's not big deal at all. So maybe this could be the way.
> 
> @pramodreddypandiri clicks aren't bad, but where could we maintain them.?

Yeah, this way people just keep upvoting their fav sites, i guess firebase authentication is better

**Comment by ManishBisht777** [comment]
## My initial idea was to add a rating on each resource 

ex - like for animation `animate on scroll`

- How beginner-friendly it is? out of 5
- How well it is documented? out of 5
- Library and framework support? out of 5 

> These are just example questions

And Everyone can rate them on these criteria

### For rating,
- We first need a auth system so that we can uniquely identify each person
> Cookies and local storage might not work

- Need a database for storing that information

### Possible tools which can handle both- `firebase and supabase` (can be more)

**Comment by Weverak** [comment]
> ## My initial idea was to add a rating on each resource 
> 
> ex - like for animation `animate on scroll`
> 
> - How beginner-friendly it is? out of 5
> - How well it is documented? out of 5
> - Library and framework support? out of 5 
> 
> > These are just example questions
> 
> And Everyone can rate them on these criteria
> 
> ### For rating,
> - We first need a auth system so that we can uniquely identify each person
> > Cookies and local storage might not work
> 
> - Need a database for storing that information
> 
> ### Possible tools which can handle both- `firebase and supabase` (can be more) 
> 
> 

That's good. And I agree with @CBID2 . Do it as testing feature and the as official.

**Comment by CBID2** [comment]
> > > > > We could use cookies. Users don't delete them usually and you can do it easily, but cookies will expire someday. Maybe firebase authentication, but users would have to register and that's another account for them and it's not big deal at all. So maybe this could be the way.
> 
> > > > > @pramodreddypandiri clicks aren't bad, but where could we maintain them.?
> 
> > > > 
> 
> > > > 
> 
> > > > Google OAuth of Firebase right ?? so we have authentication sorted that way then are we going to use redux or something of that sort to maintain the likes or upvotes of users??
> 
> > > 
> 
> > > 
> 
> > > Yes, maybe give option for standart email. Email/password auth can work also good.
> 
> > 
> 
> > I see that people want to do this voting feature. How about we do this as a trial run before making it an official feature? That way we can determine if it brings more engagement or just fosters more gatekeeping. What do you think @rupali-codes?
> 
> 
> 
> That's a nice idea

Then, it's settled! :)

**Comment by Isha988** [comment]
@rupali-codes @CBID2 @ManishBisht777 @Weverak are you people still interested in adding this feature? If yes, so, please tell me I want to work on this.

With the authentication and rating feature, we can also automate the process of adding resources in future.

**Comment by Weverak** [comment]
> @rupali-codes @CBID2 @ManishBisht777 @Weverak are you people still interested in adding this feature? If yes, so, please tell me I want to work on this.
> 
> With the authentication and rating feature, we can also automate the process of adding resources in future.

Yes, I am still interested in it, but I don't want to work on it.

**Comment by Isha988** [comment]
@Weverak I want to work on this. I am also a Assoc contributor.

**Comment by Isha988** [comment]
Thank you @CBID2 @rupali-codes for assigning it to me. Just wanna ask what is your choice between Firebase and Appwrite for making the backend.

**Comment by CBID2** [comment]
> Thank you @CBID2 @rupali-codes for assigning it to me. Just wanna ask what is your choice between Firebase and Appwrite for the backend.

I heard good things about Appwrite, but I’ll wait for @rupali-codes‘ response.

**Comment by Isha988** [comment]
Hey @rupali-codes @CBID2 @ujjawaltyagii before starting working on this issue I want to ask two things 
1. Should I use AppWrite or Firebase?
2. Where should I store the voting count in a database or JSON files? If in the database what will the unique identifier for an entry and if in JSON so it will become bulky and inconvenient?

@Weverak @ManishBisht777 @Divyansh-0  Would you guys please help me in finding the right approach for solving this issue?

**Comment by Weverak** [comment]
I would like to help and I will try my best, but I am not so good at programming and so.

**Comment by rupali-codes** [comment]
@Isha988 let's keep this issue on hold for now, we will let you know when this issue would open again!

**Comment by rupali-codes** [comment]
I'm closing this issue because we are not working on it currently.
